Build cephcsi/cephfsplugin and k8scloudprovider/manila-csi-plugin images

gouthampacha opened the pull request:
and publish them into the local registry.

gouthampacha committed:
Build cephcsi/cephfsplugin and k8scloudprovider/manila-csi-plugin images

and publish them into the local registry.

tombarron commented on roles/master/tasks/main.yml:
```diff
@@ -1,3 +1,74 @@
+    state: directory
+
+- name: check if docker cephfs image exists
+  command: >
```
This needs to be shell module rather than command module b/c it is a pipeline.

Debug shows it running a single, successful grep command.

Or we may be able to use command module with:
     docker image inspect quay.io/cephcsi/cephfsplugin:v1.0.0

gouthampacha replied:
Oh... I thought the grep here would fail..

gouthampacha replied:
```
[centos@master ~]$ echo `docker images quay.io/cephcsi/cephfsplugin:v1.0.0` | grep -v grep | grep -w quay.io/cephcsi/cephfsplugin | grep -w v1.0.0
REPOSITORY TAG IMAGE ID CREATED SIZE quay.io/cephcsi/cephfsplugin v1.0.0 29217d82dfc7 15 hours ago 471MB
[centos@master ~]$ echo $?
0
[centos@master ~]$ echo `docker images quay.io/cephcsi/cephfsplugin:v1.0.0` | grep -v grep | grep -w quay.io/cephcsi/cephfsplugin2 | grep -w v1.0.0
[centos@master ~]$ echo $?
1
```


I can replace this with a `docker image inspect` like you suggested

tombarron replied:
There are a bunch of greps and debug shows '0' return code.  Maybe it's getting that from the first 'grep -v' command.  I'm testing with the docker inspect command (which web search indicates shoule return non-zero if it can't match and zero if it does match) now.

tombarron replied:
This worked:

- name: check if docker cephfs image exists
  command: >
     docker image inspect quay.io/cephcsi/cephfsplugin:v1.0.0
  register: result
  ignore_errors: True

- name: make the cephfs image and push it to the local registry
  become: no
  shell: |
    source /etc/profile.d/golang.sh
    go get -d github.com/ceph/ceph-csi
    cd /home/centos/go/src/github.com/ceph/ceph-csi
    git checkout master
    make image-cephfsplugin
  when: result is failed

tombarron replied:
My formatting was screwed up.  https://paste.fedoraproject.org/paste/-FmNUhe4I4~AucwZAv1qcA

tombarron replied:
We can 'become: no' here, right?

gouthampacha replied:
grr, commented elsewhere. Ack on both issues. Real response here: https://github.com/tombarron/manila-kube/pull/11#pullrequestreview-230297167

tombarron commented on roles/master/tasks/main.yml:
```diff
@@ -1,3 +1,74 @@
+  when: check_cephfs_image_is_not_present | bool
+
+- name: check if openstack-cloud-provider repository has been cloned
+  stat:
```
become: no ?

gouthampacha replied:
yes

gouthampacha reviewed (commented):
Awesome, yes. we can do as you suggested, and lower the privilege. I'll cleanup the privilege escalation by inverting our current model (use sudo only in tasks where necessary) in another PR.. 

Testing these fixes right now, will update.

tombarron commented on roles/master/tasks/main.yml:
```diff
@@ -1,3 +1,74 @@
+  when: not (cloud_provider_openstack_repo.stat.exists and cloud_provider_openstack_repo.stat.isdir)
+
+- name: check if docker manila-csi image exists
+  command: >
```
become: no

gouthampacha replied:
will do

tombarron commented on roles/master/tasks/main.yml:
```diff
@@ -1,3 +1,74 @@
+
+- name: check if docker manila-csi image exists
+  command: >
+     echo `docker images k8scloudprovider/manila-csi-plugin` | grep -v grep | grep -w k8scloudprovider/manila-csi-plugin
```
docker image inspect ... as above

gouthampacha replied:
Done

gouthampacha committed:
Change image inspection command, and de-escalate privileges

Docker images command was too long, we can use docker-inspect.
We don't need sudo to execute docker commands since "centos"
user is in the "docker" group.

tombarron reviewed (approved):
LGTM
